common/unit_test/Test_prep_transpose_view.cpp

```diff
@@ -0,0 +1,250 @@
+                "Unable to get managed input view type");
+  static_assert(!OutViewType::memory_traits::is_unmanaged,
+                "Unable to get managed output view type");
+  using LayoutType = typename InViewType::array_layout;
+
+  // no need to allocate
+  {
+    LayoutType layout;
+    KokkosFFT::axis_type<DIMS> map;
+    for (int i = 0; i < DIMS; ++i) {
+      layout.dimension[i] = 5;
+      map[i]              = i;
+    }
+    InViewType in("in", layout);
+    OutViewType out("out", layout);
+    auto data_prev = out.data();
+    KokkosFFT::Impl::_prep_transpose_view(in, out, map);
+    // ensure no allocation
+    EXPECT_EQ(data_prev, out.data());
+    // check shape
+    for (int i = 0; i < DIMS; ++i) {
+      EXPECT_EQ(out.extent(i), 5);
+    }
+  }
+  // allocate
+  {
+    LayoutType layout;
+    KokkosFFT::axis_type<DIMS> map;
+    for (int i = 0; i < DIMS; ++i) {
+      layout.dimension[i] = 5;
+      map[i]              = i;
+    }
+    InViewType in("in", layout);
+    OutViewType out;
+    KokkosFFT::Impl::_prep_transpose_view(in, out, map);
+    // check shape
+    for (int i = 0; i < DIMS; ++i) {
+      EXPECT_EQ(out.extent(i), 5);
+    }
+  }
+}
+
+template <typename InViewType, typename OutViewType>
+void test_unmanaged_prep_transpose_view() {
+  constexpr std::size_t DIMS = InViewType::rank();
+  static_assert(InViewType::rank() == OutViewType::rank(),
+                "input and output have different ranks");
+
+  using InManagedViewType =
+      typename KokkosFFT::Impl::managable_view_type<InViewType>::type;
+  using OutManagedViewType =
+      typename KokkosFFT::Impl::managable_view_type<OutViewType>::type;
+  static_assert(!InManagedViewType::memory_traits::is_unmanaged,
+                "Unable to get managed input view type");
+  static_assert(!OutManagedViewType::memory_traits::is_unmanaged,
+                "Unable to get managed output view type");
+  using LayoutType = typename InViewType::array_layout;
+
+  // no need to reshape
+  {
+    LayoutType layout;
+    KokkosFFT::axis_type<DIMS> map;
+    for (int i = 0; i < DIMS; ++i) {
+      layout.dimension[i] = 5;
+      map[i]              = i;
+    }
+    InManagedViewType in("in", layout);
+    OutManagedViewType out("out", layout);
+    OutViewType u_out(out.data(), layout);
+    auto data_prev = out.data();
+    KokkosFFT::Impl::_prep_transpose_view(in, u_out, map);
+    EXPECT_EQ(data_prev, u_out.data());
+    // check shape
+    for (int i = 0; i < DIMS; ++i) {
+      EXPECT_EQ(u_out.extent(i), 5);
+    }
+  }
+  // reshape success
+  {
+    LayoutType layout;
+    KokkosFFT::axis_type<DIMS> map;
+    for (int i = 0; i < DIMS; ++i) {
+      layout.dimension[i] = 5;
+      map[i]              = i;
+    }
+
+    LayoutType layout_orig;
+    layout_orig.dimension[0] = 1;
+    for (int i = 0; i < DIMS; ++i) {
+      layout_orig.dimension[0] *= 5;
+    }
+    InManagedViewType in("in", layout);
+    OutManagedViewType out("out", layout_orig);
```

Comment:
Seems like we need a rank-1 version of `OutManagedViewType` for this operation.

Reply:
Added, as well as properly specify all extents for higher dimensions to 1.

Reply:
Sure